src/hist/named.py

```diff
@@ -1,5 +1,80 @@
 from .core import BaseHist
+import numpy as np
 
 
```

Comment:
We may move some/most of the logic to BaseHist later, with this just wrapping it with errors if a user does not use names, but this is great for now!

Reply:
okay.